sdk/cargo-build-bpf/src/main.rs

```diff
@@ -56,18 +57,125 @@ where
     }
     println!();
 
-    let mut child = Command::new(program)
+    let child = Command::new(program)
         .args(&args)
+        .stdout(Stdio::piped())
         .spawn()
         .unwrap_or_else(|err| {
             eprintln!("Failed to execute {}: {}", program.display(), err);
             exit(1);
         });
 
-    let exit_status = child.wait().expect("failed to wait on child");
-    if !exit_status.success() {
+    let output = child.wait_with_output().expect("failed to wait on child");
+    if !output.status.success() {
         exit(1);
     }
+    output
+        .stdout
+        .as_slice()
+        .iter()
+        .map(|&c| c as char)
+        .collect::<String>()
+}
+
+// Check whether a package is installed and install it if it is missing.
+fn install_if_missing(
+    config: &Config,
+    package: &str,
+    version: &str,
+    url: &str,
+    file: &Path,
+) -> Result<(), String> {
+    // Check whether the package is already in ~/.cache/solana.
+    // Donwload it and place in the proper location if not found.
+    let home_dir = PathBuf::from(env::var("HOME").unwrap_or_else(|err| {
+        eprintln!("Can't get home directory path: {}", err);
+        exit(1);
+    }));
+    let target_path = home_dir
+        .join(".cache")
+        .join("solana")
+        .join(version)
+        .join(package);
+    if !target_path.is_dir() {
+        if target_path.exists() {
+            fs::remove_file(&target_path).map_err(|err| err.to_string())?;
+        }
+        let mut url = String::from(url);
+        url.push('/');
+        url.push_str(version);
+        url.push('/');
+        url.push_str(file.to_str().unwrap());
+        download_file(&url.as_str(), &file, false)?;
+        fs::create_dir_all(&target_path).map_err(|err| err.to_string())?;
+        let tar = PathBuf::from("tar");
```

Comment:
The `tar` crate is nice and will avoid a system `tar` dependency.  Example usage at https://github.com/solana-labs/solana/blob/4bbeb9c033e4618e8be0ce9d9ad5e7edbc90f48b/runtime/src/snapshot_utils.rs#L739-L740

Reply:
done